riscv/src/bootloader.rs

```diff
@@ -30,13 +46,13 @@ branch_if_zero x1, end_page_loop;
 
 start_page_loop::
 
-// Start address
-x3 <=X= ${{ ("bootloader_input", x2 * (256 + 1) + {num_registers} + 1) }};
-x3 <== wrap(x3);
+// Page number
+x3 <=X= ${{ ("bootloader_input", x2 * ({words_per_page} + 1) + {num_registers} + 1) }};
+x3 <== and(x3, {page_number_mask});
```

Comment:
Can we somehow avoid this `and` here? Just thinking whether this might use the `binary` machine too many times.

Reply:
Right here it's just once, per page, right? Don't think that's a problem.